Source/main.cpp

```diff
@@ -18,50 +18,60 @@
   */
 /* USER CODE END Header */
 /* Includes ------------------------------------------------------------------*/
+#include <stdint.h>
+#include <time.h>
 #include "main.h"
-#include "adc.h"
-
 #include "hydrophones.h"
-#include <stdint.h>
-
 #include "stm32f7xx_hal.h"
+#include "adc.h"              /* Somehow throws an error */
 
 /* USER CODE END Includes */
 
 /* Private variables ---------------------------------------------------------*/
 
-// Handler for the ADC and the DMA
-ADC_HandleTypeDef hadc1;
-DMA_HandleTypeDef hdma_adc;
+/* Handler for the ADC, DMA, ETH and SPI */
+ADC_HandleTypeDef hadc1;      /* Somehow doesn't exists */
```

Comment:
Do these exist now?

Reply:
![image](https://user-images.githubusercontent.com/47692941/103532437-88f35080-4e8b-11eb-83a8-b40dd9e5c949.png)

The compiler cannot find them, even though the code/definition is produced by someone competent at ARM. Must be an include-error somewhere that I cannot find, as even the.c file containing the decleration throws an error...

Reply:
The error seems to be resolved. Looked like it was caused by a combination of forgetting to include at least one file, and circular-dependencies in the files that were included